src/poliastro/twobody/events.py

```diff
@@ -25,8 +30,84 @@ def terminal(self):
     def last_t(self):
         return self._last_t * u.s
 
-    def __call__(self, t, u, k):
+    def __call__(self, t, u_, k):
         self._last_t = t
-        H = norm(u[:3])
+        H = norm(u_[:3])
         # SciPy will search for H - R = 0
         return H - self._R
+
+
+class PenumbraEvent:
+    """Detect whether a satellite is in penumbra or not.
+
+    Parameters
+    ----------
+    r_sun: ~astropy.units.Quantity
+        Position vector of the Sun with respect to the Earth.
+    R: ~astropy.units.Quantity
+        Radius of the attractor.
+
+    """
+
+    def __init__(self, r_sun, r_s, r_p, terminal=False):
```

Comment:
Thanks again. Since `r_sun` is the position of secondary body wrt primary body, it might not be related to the orbit? So it would need to be passed as a separate argument, right?

*(`r_sun` is changed to `r_sec`)*